lib/puppet/gettext_config.rb

```diff
@@ -0,0 +1,85 @@
+module Puppet::GettextConfig
+  LOCAL_PATH = File.absolute_path('../../locales', File.dirname(__FILE__))
+  POSIX_PATH = File.absolute_path('../../../../share/locale', File.dirname(__FILE__))
+  WINDOWS_PATH = File.absolute_path('../../../../../../puppet/share/locale', File.dirname(__FILE__))
+
+  # Search for puppet gettext config files
+  # @return path to the config, or nil if not found
+  def self.puppet_locale_path
+    if File.exist?(LOCAL_PATH)
+      return LOCAL_PATH
+    elsif Puppet::Util::Platform.windows? && File.exist?(WINDOWS_PATH)
```

Comment:
we should probably explicitly `require puppet/util/platform` since we use it here. we're inconsistent about where we just rely on dependencies being transitively satisfied but I lean towards being explicit if possible